plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1431,6 +1432,7 @@ def build_transaction(  # pylint: disable=too-many-arguments
             if gas_data:
                 tx_params.update(gas_data)  # pragma: nocover
         tx = tx.buildTransaction(tx_params)
+        tx = self.update_with_gas_estimate(tx)
```

Comment:
I think the problem is now that it's impossible to provide user defined gas, gas estimation will always take precedence.

Also see

https://github.com/valory-xyz/open-aea/pull/269/files#diff-26c50f98b25886126548a663e3e23b7ab5f34c3599671d81e0f3efc78cf0121fL1348-L1352


I think in both places, we should at least have symmetry

Reply:
E.g. here
```suggestion
        if self._is_gas_estimation_enabled:
            tx = self.update_with_gas_estimate(tx)
```